modules/cloudiot/main.tf

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_cloudiot_registry" "default" {
+  name    = var.name
+  project = var.project_id
+  region  = var.region
+
+  http_config = {
+    http_enabled_state = var.http_enabled_state
+  }
+  mqtt_config = {
+    mqtt_enabled_state = var.mqtt_enabled_state
+  }
+  event_notification_configs {
```

Comment:
Should this be done as a dynamic block, so it's not even created if `event_notification_config.topic` is unset?